tests/integration/backward_compatible/proxy/map_test.py

```diff
@@ -1003,3 +1015,40 @@ def test_min_with_attribute_path(self):
     def test_min_with_predicate(self):
         average = self.map.aggregate(min_(), greater_or_equal("this", 3))
         self.assertEqual(3, average)
+
+
+@unittest.skipIf(
+    is_client_version_older_than("4.2.1"), "Tests the features added in 4.2.1 version of the client"
+)
+class MapProjectionsTest(SingleMemberTestCase):
+    @classmethod
+    def configure_client(cls, config):
+        config["cluster_name"] = cls.cluster.id
+        return config
+
+    def setUp(self):
+        self.map = self.client.get_map(random_string()).blocking()
+        self.map.put(1, HazelcastJsonValue('{"attr1": 1, "attr2": 2, "attr3": 3}'))
```

Comment:
lets put one more value to map to verify that predicates work. You might need to update the tests accordingly